src/Common/RadixSort.h

```diff
@@ -529,6 +564,38 @@ struct RadixSort
         radixSortLSDInternal<true>(arr, size, reverse, destination);
     }
 
+    /** Tries to fast sort elements for common sorting patterns (unstable).
+      * If fast sort cannot be performed, execute least significant digit radix sort.
+      */
+    static void executeLSDWithTrySort(Element * arr, size_t size)
+    {
+        return executeLSDWithTrySort(arr, size, false);
+    }
+
+    static void executeLSDWithTrySort(Element * arr, size_t size, bool reverse)
+    {
+        return executeLSDWithTrySort(arr, size, reverse, nullptr);
+    }
+
+    static void executeLSDWithTrySort(Element * arr, size_t size, bool reverse, Result * destination)
```

Comment:
This seems a very error prone API. `arr` might be modified, even if destination is passed, which is unexpected. It's better to remove it and leave only the in-place sort, IMO.

Reply:
There is function with such API:
```
    /** This function will start to sort inplace (modify 'arr')
      *  but on the last step it will write result directly to the destination
      *  instead of finishing sorting 'arr'.
      * In this case it will fill only Result parts of the Element into destination.
      * It is handy to avoid unnecessary data movements.
      */
    static void executeLSD(Element * arr, size_t size, bool reverse, Result * destination)
    {
        radixSortLSDInternal<true>(arr, size, reverse, destination);
    }
```
This API is great for RadixSort, because a lot of the time you need to construct temporary array with index and value. Temporary array will be sorted by values, but in result you only need indexes, and temporary array can be dropped.

Reply:
:ok_hand: